src/components/Contexts/FormDataProvider.jsx

```diff
@@ -48,8 +49,9 @@ const FormDataProvider = ({ children }) => {
           metadata: { qualification: newQualification },
           dirId: appFolderID
         })
-        setIsStepperDialogOpen(false)
       }
+      setIsStepperDialogOpen(false)
```

Comment:
Shouldn't we await the response of the previous function call? What if it's unsuccessful?

Reply:
It seems alright.